src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -890,9 +890,24 @@ private void RetrieveBoundOperationPaths(OpenApiConvertSettings convertSettings)
 
                 foreach (var navProperty in returnBindingEntityType.NavigationProperties())
                 {
-                    ODataPath newNavigationPath = functionPath.Clone();
-                    newNavigationPath.Push(new ODataNavigationPropertySegment(navProperty));
-                    AppendPath(newNavigationPath);
+                    if (convertSettings.ComposableFunctionsExpansionDepth > 0)
+                    { // Generate nav. props of composable functions
+                        ODataSegment secondLastSeg = functionPath.ElementAt(functionPath.Count - 2);
+
+                        if (convertSettings.ComposableFunctionsExpansionDepth < 2 &&
+                            functionPath.LastSegment is ODataOperationSegment &&
+                            secondLastSeg is ODataOperationSegment)
+                        {
+                            // Only one level of composable functions expansion allowed
+                            continue;
```

Comment:
Does this mean that we cannot introduce granularity like limiting the path to 3 or 4 expansions?

Reply:
Good feedback. I have made the generation more granular, depending on the setting